File: fortran/example_mHT_optional_parameters.f90
program example_mHT_optional_parameters
   use, intrinsic :: iso_fortran_env, only: int32, dp => real64
   use spectral_module, only: mHTprofile
   implicit none
   real(dp) :: absorption, dispersion
! Example parameters of the S(1) 3-0 line of H2 perturbed by He (reference: 10.1103/PhysRevA.101.052705)
   real(dp) :: nu0 = 112265.5949_dp ! Unperturbed line position in cm-1.
   real(dp) :: GammaD = 35.1e-3_dp ! Doppler broadening in cm-1.
   real(dp) :: Gamma0 = 11.7e-3_dp ! Speed-averaged line-width in cm-1.
   real(dp) :: Gamma2 = 5.4e-3_dp ! Speed dependence of the line-width in cm-1.
   real(dp) :: Delta0 = 30.5e-3_dp ! Speed-averaged line-shift in cm-1.
   real(dp) :: Delta2 = 12.4e-3_dp ! Speed dependence of the line-shift in cm-1.
   real(dp) :: NuOptRe = 38.0e-3_dp ! Real part of the Dicke parameter in cm-1.
   real(dp) :: NuOptIm = -17.5e-3_dp ! Imaginary part of the Dicke parameter in cm-1.
   real(dp) :: nu ! Current wavenumber of the computation in cm-1.
   real(dp) :: Ylm ! Real part of the 1st order (Rosenkranz) line mixing coefficients, dimensionless.
   real(dp) :: Xlm ! Imaginary part of the 1st order (Rosenkranz) line mixing coefficients, dimensionless.
   real(dp) :: alpha ! perturber-to-absorber mass ratio, dimensionless.
   logical  :: calculate_dispersion  ! The mHT function will return either the dispersion or absorption profile.
   
   nu  = nu0 + 1.0_dp
   absorption = mHTprofile(nu0,GammaD,Gamma0,Gamma2,Delta0,Delta2,     &
      NuOptRe,NuOptIm,nu)
   calculate_dispersion = .true.
   dispersion = mHTprofile(nu0,GammaD,Gamma0,Gamma2,Delta0,Delta2,     &
      NuOptRe,NuOptIm,nu,calculate_dispersion_opt=calculate_dispersion)
   write(*, '(A, 25X, 2F22.15)') "The output of the mHT function "     &
      // "(He-perturbed S(1) 3-0 line in H2):", absorption, dispersion
! optional parameters
   Ylm = 1.0e-3_dp
   Xlm = 0.5e-3_dp
   alpha = 2.0_dp

   absorption = mHTprofile(nu0,GammaD,Gamma0,Gamma2,Delta0,Delta2,     &
      NuOptRe,NuOptIm,nu,Ylm,Xlm,alpha)
   dispersion = mHTprofile(nu0,GammaD,Gamma0,Gamma2,Delta0,Delta2,     &
      NuOptRe,NuOptIm,nu,Ylm,Xlm,alpha,calculate_dispersion)

   write(*, '(A, 2F22.15)') "The output of the mHT function with "     &
      // "optional parameters (He-perturbed S(1) 3-0 line in H2):",    &
      absorption, dispersion
end program example_mHT_optional_parameters

File: Python/example_mHT_optional_parameters.py
try: 
  from mHT import mHTprofile
except ImportError as msg: 
  raise SystemExit (str(msg) + '\nexample_mHT_optional_parameters.py: mHT not found. Make sure catalog with mHT module is in same catalog as running script and is named "mHT"!')

# Example parameters of the S(1) 3-0 line of H2 perturbed by He (Source: 10.1103/PhysRevA.101.052705)
nu0     = 112265.5949 # Unperturbed line position in cm-1.
GammaD  = 35.1e-3     # Doppler broadening in cm-1.
Gamma0  = 11.7e-3     # Speed-averaged line-width in cm-1.
Gamma2  = 5.4e-3      # Quadratic speed dependence parameter of the line-width in cm-1.
Delta0  = 30.5e-3     # Speed-averaged line-shift in cm-1.
Delta2  = 12.4e-3     # Quadratic speed dependence parameter of the line-shift in cm-1.
NuOptRe = 38.0e-3     # Real part of the Dicke parameter in cm-1.
NuOptIm = -17.5e-3    # Imaginary part of the Dicke parameter in cm-1.
nu      = nu0+1       # Current wavenumber in cm-1.

print("The output of the mHT function:")
print(mHTprofile(nu0,GammaD,Gamma0,Gamma2,Delta0,Delta2,NuOptRe,NuOptIm,nu)) # The mHT function output returns absorption profile by default. To display the dispersive profile see example_dispersion.py.

# Optional parameters
newXlm   = 1.0e-3 # Real part of the 1st order (Rosenkranz) line mixing coefficients, dimensionless (default: 0.0).
newYlm   = 0.5e-3 # Imaginary part of the 1st order (Rosenkranz) line mixing coefficients, dimensionless (default: 0.0).
newalpha = 2      # Perturber-to-absorber mass ratio, dimensionless (default: 10.0).
newdisp  = True   # Boolean trigger for including dispersion profile in the output (default: False).

print("The output of the mHT function with optional parameters:")
print(mHTprofile(nu0,GammaD,Gamma0,Gamma2,Delta0,Delta2,NuOptRe,NuOptIm,nu,newYlm,newXlm,newalpha,newdisp))
# This function accepts 4 optional parameters.
# Parameters are not key-worded, so they need to be specified in specific order,
# e.g. to specify dispersion parameter all other parameters has to be specified as well.
